core/src/main/java/feign/RetryableException.java

```diff
@@ -47,6 +47,16 @@ public RetryableException(int status, String message, HttpMethod httpMethod, Dat
     this.retryAfter = retryAfter != null ? retryAfter.getTime() : null;
   }
 
+  /**
+   * @param retryAfter usually corresponds to the {@link feign.Util#RETRY_AFTER} header.
+   */
+  public RetryableException(String message, HttpMethod httpMethod, Throwable cause,
+      Date retryAfter, Request request) {
+    super(500, message, request, cause);
+    this.httpMethod = httpMethod;
+    this.retryAfter = retryAfter != null ? retryAfter.getTime() : null;
```

Comment:
This exception is meant to portray a situation where the `Retry-After` header is present.  Given that, it's status should be taken from the use case.  Forcing it to 500 is incorrect in this use case.